src/apply.c

```diff
@@ -291,7 +291,15 @@ static int apply_binary(
 	git_patch *patch)
 {
 	git_buf reverse = GIT_BUF_INIT;
-	int error;
+	int error = 0;
+
+	if (!patch->binary.old_file.datalen && !patch->binary.new_file.datalen)
+		goto done;
+
+	if (!patch->binary.contains_data) {
+		error = apply_err("patch does not contain binary data");
+		goto done;
+	}
```

Comment:
Can this even be triggered? If `contains_data == 0`, we'll usually have `old_file.datalen == 0 && new_file.datalen == 0` due to the patch not being parsed. So actually, we'd already skip this check due to the previous check. I'd rather expect that this statement and the previous statement needs to be swapped.